.github/workflows/test/issues.test.js

```diff
@@ -0,0 +1,98 @@
+import { describe, expect, it, vi, beforeEach } from "vitest";
+import { getIssueNumber } from "../src/issues.js";
+import { createMockGithub, createMockCore } from "./mocks.js";
+
+const mockGithub = createMockGithub();
+const mockCore = createMockCore();
+
+describe("getIssueNumber", () => {
+  beforeEach(() => {
+    // Reset mocks
+    vi.clearAllMocks();
+  });
+  it("should return NaN when head_sha is missing", async () => {
```

Comment:
Should this throw instead of returning NaN?  Is it ever valid to call without a head_sha?